spec/models/accounting/transaction_spec.rb

```diff
@@ -122,4 +121,46 @@
       expect(Accounting::Transaction.standard_order).to eq([txn_4, txn_5, txn_3, txn_2, txn_1])
     end
   end
+
+  context 'with line items' do
+    before do
+      create(:line_item, parent_transaction: transaction, posting_type: 'debit',
+        account: transaction.division.principal_account, amount: 1)
+      create(:line_item, parent_transaction: transaction, posting_type: 'debit',
+        account: transaction.division.interest_receivable_account, amount: 2)
+      create(:line_item, parent_transaction: transaction, posting_type: 'debit',
+        account: transaction.division.interest_income_account, amount: 1.5)
+      create(:line_item, parent_transaction: transaction, posting_type: 'debit',
+        amount: 2.5)
+      create(:line_item, parent_transaction: transaction, posting_type: 'credit',
+        account: transaction.division.principal_account, amount: 5)
+      create(:line_item, parent_transaction: transaction, posting_type: 'credit',
+        account: transaction.division.interest_receivable_account, amount: 3)
+      create(:line_item, parent_transaction: transaction, posting_type: 'credit',
+        account: transaction.division.interest_income_account, amount: 1)
+      create(:line_item, parent_transaction: transaction, posting_type: 'credit',
+        amount: 11)
+    end
+
+    it 'calculates correctly' do
+      expect(transaction.reload.change_in_principal).to eq(-4)
+      expect(transaction.reload.change_in_interest).to eq(-1)
```

Comment:
Nice test! I'd suggest adjusting a little bit so that the result is not a whole number. (Just add a penny someplace). Else the result might be getting truncated somewhere and you wouldn't know.